Контрольная работа 1.3 задание 3/main.c

```diff
@@ -0,0 +1,102 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+#define MAX_LENGTH 100
+
+void deleteCopies(char string[])
+{
+    char* newString = string;
+    char* example = string + '\0';
+
+    while (*example != NULL)
+    {
+        if (*example != *newString)
+        {
+            ++newString;
+            *newString = *example;
+        }
+        ++example;
+    }
+
+    *(++newString) = '\0';
+}
+
+bool testing()
+{
+    bool flag = true;
+    FILE* file1 = NULL;
+    fopen_s(&file1, "test1.txt", "r");
+    if (file1 == NULL)
+    {
+        flag = false;
+    }
+
+    char string1[MAX_LENGTH] = { '\0' };
+    if (fscanf_s(file1, "%s", string1, MAX_LENGTH) != 1)
+    {
+        flag = false;
+    }
+
+    deleteCopies(string1);
+    if (strcmp(string1, "ababababab") != 0)
+    {
+        flag = false;
+    }
+
+    fclose(file1);
+
+    FILE* file2 = NULL;
+    fopen_s(&file2, "test2.txt", "r");
+    if (file2 == NULL)
+    {
+        flag = false;
+    }
+
+    char string2[MAX_LENGTH] = { '\0' };
+    if (fscanf_s(file2, "%s", string2, MAX_LENGTH) != 1)
+    {
+        flag = false;
+    }
+
+    deleteCopies(string2);
+    if (strcmp(string2, "abababa\qwewrty") != 0) // the function doesn't work with line breaks, so the testing will be failed
```

Comment:
```suggestion
    if (strcmp(string2, "abababa\nqwewrty") != 0) // the function doesn't work with line breaks, so the testing will be failed
```
strcmp без проблем работает с переводами строк, а вот fscanf_s считает строку до разделителя (перевода строки или даже пробела), так что тест действительно не пройдёт. Но потому, что задача решена неправильно, надо было fgetc-ом файл читать :)